crates/nu-protocol/src/value/mod.rs

```diff
@@ -806,28 +806,15 @@ impl Value {
                 Type::Record(val.iter().map(|(x, y)| (x.clone(), y.get_type())).collect())
             }
             Value::List { vals, .. } => {
-                let mut ty = None;
-                for val in vals {
-                    let val_ty = val.get_type();
-                    match &ty {
-                        Some(x) => {
-                            if &val_ty != x {
-                                if x.is_numeric() && val_ty.is_numeric() {
-                                    ty = Some(Type::Number)
-                                } else {
-                                    ty = Some(Type::Any);
-                                    break;
-                                }
-                            }
-                        }
-                        None => ty = Some(val_ty),
-                    }
-                }
+                let ty = vals
+                    .iter()
+                    .map(Value::get_type)
+                    .reduce(|e, acc| e.widen(acc));
```

Comment:
Is there a benchmark for this kind of tasks?

This could be potentially time consuming.